atlite/hydro.py

```diff
@@ -103,6 +103,13 @@ def shift_and_aggregate_runoff_for_plants(
         nhours = (distances * (flowspeed * 3.6) + 0.5).astype(int)
 
         for b in ppl.upstream:
-            inflow_plant += runoff.sel(hid=b).shift(time=nhours.at[b])
+            # get upstream inflows
+            upstream_inflow_b = runoff.sel(hid=b).shift(time=nhours.at[b])
+            # replace null values with zeros
+            upstream_inflow_b = upstream_inflow_b.where(
+                upstream_inflow_b.isnull() == False, 0.0
+            )
+            # update inflow for the plant
+            inflow_plant += upstream_inflow_b
```

Comment:
```suggestion
            upstream_inflow_b = runoff.sel(hid=b).shift(time=nhours.at[b], fill_value=0)
```
wouldn't this already do it?

Reply:
Thank you! Much more elegant